File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    // Prompts user for non-negative float
    float dollars;
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);


    // Convert dollars to cents
    int cents = round(dollars * 100);
    printf("%i\n", cents);

    int counter = 0;

    while (cents > 0)
    {
        counter += (cents / 25);
        cents = (cents % 25);

        counter += (cents / 10);
        cents = (cents % 10);

        counter += (cents / 5);
        cents = (cents % 5);

        counter += (cents / 1);
        cents = (cents % 1);

    }
    printf("%i\n", counter);
}

File: pset6/cash/cash.py
from cs50 import get_float

while True:
    change = get_float("Change owed: ")
    if change >= 0:
        break

# Converts dollars to cents
change = round(change * 100)

# Initialise coin counter
coins = 0

while change > 0:
    # Count quarters
    coins += change // 25
    change %= 25

    # Count dimes
    coins += change // 10
    change %= 10

    # Count nickels
    coins += change // 5
    change %= 5

    # Count pennies
    coins += change // 1
    change %= 1

print(coins)